iscsi/multipath.go

```diff
@@ -8,11 +8,14 @@ import (
 	"os/exec"
 	"strings"
 	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/klog/v2"
 )
 
 // ExecWithTimeout execute a command with a timeout and returns an error if timeout is excedeed
-func ExecWithTimeout(command string, args []string, timeout time.Duration) ([]byte, error) {
-	debug.Printf("Executing command '%v' with args: '%v'.\n", command, args)
+func ExecWithTimeout(logger *logr.Logger, command string, args []string, timeout time.Duration) ([]byte, error) {
```

Comment:
This looks like a candidate for adding a ctx, in particular when the code below then has to use context.Background to create one.

Reply:
I'm reluctant to pass in context to this function; it would require that essentially all functions take a context, based on the call tree. If highly desired, I can make the change.

Reply:
I understand the reluctance, but I still think it would be the better API for this function. Just my two cents.

Reply:
Do those other functions get modified to take a klog.Logger? If yes, then the API already changes and a context also there would make more sense.

Reply:
I went ahead and made the change to use context.